muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/chat/chat.tsx

```diff
@@ -209,6 +237,123 @@ class Chat extends React.Component<IChatProps, IChatState> {
       this.setUserAvailabilityDropdown.bind(this);
   }
 
+  /**
+   * loadStudentCouncelors
+   */
+  async loadStudentCouncelors() {
+    try {
+      const studentsUserGroups: GuiderUserGroupListType = (await promisify(
+        mApi().usergroup.groups.read({
+          userIdentifier: this.props.status.userSchoolDataIdentifier,
+        }),
+        "callback"
+      )()) as GuiderUserGroupListType;
+
+      const studentsGuidanceCouncelors: SummaryStudentsGuidanceCouncelorsType[] =
+        [];
+      if (studentsUserGroups && studentsUserGroups.length) {
+        const councelGroups = studentsUserGroups.filter(
+          (studentsUserGroup) => studentsUserGroup.isGuidanceGroup == true
+        );
+
+        await Promise.all(
+          councelGroups.map(async (studentsUserGroup) => {
+            await promisify(
+              mApi().usergroup.groups.staffMembers.read(studentsUserGroup.id, {
+                properties:
+                  "profile-phone,profile-appointmentCalendar,profile-whatsapp,profile-vacation-start,profile-vacation-end",
+              }),
+              "callback"
+            )().then((result: SummaryStudentsGuidanceCouncelorsType[]) => {
+              result.forEach((studentsGuidanceCouncelor) => {
+                if (
+                  !studentsGuidanceCouncelors.some(
+                    (existingStudentCouncelor) =>
+                      existingStudentCouncelor.userEntityId ==
+                      studentsGuidanceCouncelor.userEntityId
+                  )
+                ) {
+                  studentsGuidanceCouncelors.push(studentsGuidanceCouncelor);
+                  studentsGuidanceCouncelors.sort((x, y) => {
+                    const a = x.lastName.toUpperCase(),
+                      b = y.lastName.toUpperCase();
+                    return a == b ? 0 : a > b ? 1 : -1;
+                  });
+                }
```

Comment:
You can sorting after forEach is done and before setState. Every loop is unnecessary.